src/clr-core/common/utils/aria-live.service.ts

```diff
@@ -0,0 +1,124 @@
+   * DOM Element is existing if not it will create one for us and the will apply the text.
+   *
+   * ```typescript
+   * this.ariaLiveService.announce(this.el.nativeElement);
+   * // or
+   * this.ariaLiveService.announce('Message to announce to SR');
+   * ```
+   *
+   * @remark
+   * When second argument is `AriaLivePoliteness.off` we won't create aria container or update it.
+   * The reason for that is that we don't want to do additional work if the SR will ignore it.
+   *
+   * @param message - This could be simple string or HTMLElement
+   * @param politeness - 'polite', 'assertive' or 'off'
+   */
+  announce(message: string | HTMLElement, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): void {
+    if (!message || politeness === AriaLivePoliteness.off) {
+      return;
+    }
+
+    if (typeof message !== 'string' && this.isBrowser() && message.textContent) {
+      message = message.textContent;
+    }
+
+    // when there is no message do NOTHING!
+    if (!message) {
+      return;
+    }
+
+    if (!this.ariaLiveElement && this.isBrowser()) {
+      this.ariaLiveElement = this.createContainer();
+      this.ariaLiveElement.setAttribute('aria-live', politeness);
+      this.addTextToAriaLiveElement(message);
+    }
+  }
+
+  cleanup() {
+    if (this.isBrowser() && this.ariaLiveElement) {
+      this.document.body.removeChild(this.ariaLiveElement);
+      this.ariaLiveElement = null;
+    }
+  }
+
+  protected addTextToAriaLiveElement(message: string | HTMLElement) {
+    // This clearTimeout will stop all older messages from announcing
+    // in the case where the messages are comming too fast we gonna try to append only
+    // the last one. That's what the SR will try to do anyway.
+    clearTimeout(this.previousTimeout);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    this.previousTimeout = setTimeout(() => {
+      if (this.ariaLiveElement) {
+        this.ariaLiveElement.textContent = <string>message;
+      }
+    }, ARIA_LIVE_TICK);
+  }
+
+  protected isBrowser() {
+    return typeof window !== 'undefined';
+  }
+
+  /**
+   * Create AriaLive DOM element as a last child of the document.
+   * After the element is created, we gonna apply Clarity class to hide it from
+   * the screen and set the `aria-live` politness.
+   *
+   * `clr-sr-only` is the CSS class that is used to hide the element from the screen.
+   *
+   * @remark
+   * Calling this method multiple times will create multiple DOM Elements, that
+   * won't be tracked and will be GC after the service is destroyed.
```

Comment:
This method was a bit simplified when it comes to checking in favor of be only callable from inside other internal methods that will prevent it from been called when the context is not browser. For that reason this method is working directly with DOM method. And its more or less extracting it into function for simplicity and for the rare case of been called from other method. 

Now when I look at it one or two ifs won’t make a big difference.

In its current version if called it will create another and another element with the same ID. But we won’t cleanup all of them inside the cleanup method. The comment above is more of explaining what could be the side effect. 

There could be change to remove the previous element before creating new one.

Reply:
I'd very much be in favor of writing the few lines of code it would take to make `cleanup` remove all the DOM elements and not just the last one used.

Reply:
i prefer to be sure that there no double element with the same id. 

There is no case at the moment where the same instance will have need of having more than one aria live dom element ref.

If this is the case there will be no need of searching or removing more than one element.

Reply:
I'll be reverting the aria-live service changes in this PR. We are looking to de-couple aria-live service from components themselves because a lot of the logic is dependent on application context which the component library has no visibility or access to.